connectivity/FEATURE_BLE/source/generic/SecurityManagerImpl.cpp

```diff
@@ -468,13 +662,21 @@ ble_error_t SecurityManager::setLinkSecurity(
     SecurityMode_t securityMode
 )
 {
-    if (!_db) return BLE_ERROR_INITIALIZATION_INCOMPLETE;
+    tr_info("Set %d link security %s", connection, to_string(securityMode));
+
+    if (!_db) {
+        tr_error("Failure, DB not initialized");
+        return BLE_ERROR_INITIALIZATION_INCOMPLETE;
+    }
+
     ControlBlock_t *cb = get_control_block(connection);
     if (!cb) {
+        tr_error("Failure, control block not found for connection %d", connection);
         return BLE_ERROR_INVALID_PARAM;
     }
 
     if (cb->encryption_requested) {
+        tr_error("Failure, encryption already requested");
```

Comment:
wouldn't class it as an error, since it's already pending you will actually get what you wnted

Reply:
actually no, you could ask for something more